tensordict/prototype/fx.py

```diff
@@ -40,7 +40,7 @@ def forward(
 
         for out_key, output in zip(self.out_keys, outputs):
             if out_key != "_":
-                tensordict_out.set(out_key, output, _run_checks=False)
+                tensordict_out.set(out_key, output)
```

Comment:
This change is a good example of where we would like to be able to call `_set` with nested keys support, as the runtime checks in the forward pass are very bad.

Reply:
Oh I see.
What if we duplicate the logic:
```python
def set(self, key, value, inplace=False):
    key = self._validate_key(key)
    if isinstance(key, tuple):
        td, subkey = _get_leaf_tensordict(self, key)
        td.set(subkey, value)  # value validation happens in context of leaf
        return self
    value = self._validate_value(value)
    return self._set(key, value, inplace=inplace)
def _set(self, key, value, inplace=False):
    if isinstance(key, tuple):
        td, subkey = _get_leaf_tensordict(self, key)
        td._set(subkey, value)  # value validation happens in context of leaf
        return self
    return self._set(key, value, inplace=inplace)
```

Reply:
I think that might be the best approach. It's not ideal, but I don't see a better alternative.